### PR TITLE
Make the Via form on the homepage work as a plain HTML form

### PR DESCRIPTION
**Pull request by robertknight**
This enables the Via link to do something useful even
if the homepage JS has not loaded.

A downside is that this introduces an additional redirect on the happy path. We could remove that but I'm unsure whether it is going to matter.

### CODE REVIEW

**Comment by nickstenning** [comment]
Why not keep the `onsubmit` handler (although possibly put it somewhere more sensible) so that the current behaviour is retained if JS loads and it still works if it doesn't.

**Comment by robertknight** [comment]
Relevant discussion from Slack: https://hypothes-is.slack.com/archives/public/p1450352404000009
TL;DR - We'll retain the client-side form handling to avoid the redirect for the majority of users.

File: h/home.py [hunk 1]
```diff
@@ -1,5 +1,14 @@
 """The home/front page of the //hypothes.is/ site."""
 from pyramid import view
+from pyramid import httpexceptions
+
+
+@view.view_config(route_name='via_redirect',
+                  request_method='GET')
+def via_redirect(context, request):
+    url = request.params['url']
```
Comment (nickstenning) [inline h/home.py]: This is a `KeyError` if someone visits without the `url` param. Raise a 404 or perhaps a 400 instead.

File: h/static/scripts/site.js [hunk 2]
```diff
@@ -23,7 +22,7 @@ document.addEventListener('DOMContentLoaded', function () {
   // setup route
   var route = document.location.pathname;
   if (route.match('^/(new-homepage)?$')) {
```
Comment (nickstenning) [inline h/static/scripts/site.js]: Could you remove the `new-homepage` option from this. Happy for that to be a separate commit/PR.

**Comment by judell** [comment]
This is /really/ helpful, thanks!

File: h/static/scripts/site.js (hunk 2 above)
Comment (robertknight) [inline h/static/scripts/site.js]: Done

**Comment by nickstenning** [comment]
Great!
